src/compiler/crystal/syntax/ast.cr

```diff
@@ -339,6 +339,10 @@ module Crystal
       new(values.map { |value| (yield value).as(ASTNode) }, of: of)
     end
 
+    def self.map_with_index(values, of = nil)
```

Comment:
The `of` argument is never used. Is it necessary?

Reply:
It's used on the next line `new(..., of: of)`

Reply:
I mean it's never used as an argument to this method. You could just call `new(..., of: nil)`.

Reply:
Ah, gotcha